json-rpc/src/methods.rs

```diff
@@ -356,29 +365,32 @@ async fn get_network_status(service: JsonRpcService, _request: JsonRpcRequest) -
 /// Builds registry of all available RPC methods
 /// To register new RPC method, add it via `register_rpc_method!` macros call
 /// Note that RPC method name will equal to name of function
+#[allow(unused_comparisons)]
 pub(crate) fn build_registry() -> RpcRegistry {
     let mut registry = RpcRegistry::new();
-    register_rpc_method!(registry, "submit", submit, 1);
-    register_rpc_method!(registry, "get_metadata", get_metadata, 1);
-    register_rpc_method!(registry, "get_account", get_account, 1);
-    register_rpc_method!(registry, "get_transactions", get_transactions, 3);
+    register_rpc_method!(registry, "submit", submit, 1, 0);
```

Comment:
I feel it make it more readable to design it as  total params & optional params  instead.   otherwise it is really hard to see how many arguments 

    register_rpc_method!(registry, "submit", submit, 1, 1);
    register_rpc_method!(registry, "get_metadata", get_metadata, 1, 0);
    get_transactions", get_transactions, 3, 3)

Reply:
To me, required arguments size and optional arguments size are metadata of the method, hence prefer to set them directly.
It probably does not matter as long as we pass 2 numbers, someone will feel confusing.
So you always need to read register_rpc_method macro to understand what they are.

What I'd like to add later is something like: Argument{"fieldName": {"Desc": "...", "required": true/false, "type": ....}}
Then we can generate our document from this definition, or we should generate register code from a definition that can be used to generate document.

Reply:
ok, if we could generate these from an rust Request struct with serde-json tags that would be awesome (we can even set "rename" through serde tags)